### PR TITLE
feat: Add support for generic PGVector vector store

### LINKED ISSUE #2478: PGVector integration for Langchain JS similar to Python library
Hey Guys, 
Any roadmap for support PGVector integration for the javascript lib similar to the Python lib.
Lack of this integration is quite annoying. 

One workaround is to use Supabase vector store, but frankly the Supabase integration is quite hackie. It is a considerable deviation from the python version of the PG Vector store even in terms of the table structure. It take quite a bit of work to move from python PGVector to Supabase client. 

In my opinion, the Supabase integration doesn't work well even with Python library. 

Moreover if I don't want to use Supabase, I don't see a good workaround. 

Would be grateful if this can be supported soon. 

Thanks :)

### PR DESCRIPTION
**Pull request by bracesproul**
Closes #2478 

- Adds generic PGVector vector store
- Test coverage for PGVector vector store

Run integration tests with `yarn test:single langchain/src/vectorstores/tests/pgvector.int.test.ts`

Twitter [@bracesproul](https://twitter.com/bracesproul)

### CODE REVIEW

File: langchain/src/vectorstores/pgvector.ts [hunk 1]
```diff
@@ -0,0 +1,300 @@
+import { Metadata } from "@opensearch-project/opensearch/api/types.js";
```
[the remark this answers is not in the document]
Comment (bracesproul) [reply langchain/src/vectorstores/pgvector.ts]: Accidental import, resolved in recent commit.

**Comment by jacoblee93** [comment]
This is fantastic and much needed, thanks! 

Made a few small fixes to the `end()` method to get tests passing but will merge tonight.

**Comment by bracesproul** [comment]
> This is fantastic and much needed, thanks!
> 
> Made a few small fixes to the `end()` method to get tests passing but will merge tonight.

tyvm! That `.end()` had me super confused and I had assumed it was an issue with my local setup haha. Thanks for pushing up those fixes!

**Comment by jacoblee93** [comment]
No worries, I also added the image to our integration tests.

And thank you for using prepared statements 🙏

**Comment by rahulnyk** [comment]
This is awesome!
Out of curiosity, Does this push support MMR search with PGVector store?

**Pull request merged**